common/src/main/java/org/dash/wallet/common/payments/parsers/AddressParser.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.dash.wallet.common.payments.parsers
+
+import org.bitcoinj.core.Address
+import org.bitcoinj.core.Base58
+import org.bitcoinj.core.NetworkParameters
+
+open class AddressParser(pattern: String, val params: NetworkParameters?) {
+    companion object {
+        val PATTERN_BITCOIN_ADDRESS = "[${Base58.ALPHABET.joinToString(separator = "")}]{20,40}"
```

Comment:
This was moved from `wallet` but changed to a class that can handle different patterns.